modules/tinymce/src/core/main/ts/newline/InsertBlock.ts

```diff
@@ -387,19 +390,44 @@ const insert = (editor: Editor, evt?: EditorEvent<KeyboardEvent>): void => {
     );
     editor.selection.setCursorLocation(newBlock, 0);
   } else if (CaretContainer.isCaretContainerBlock(parentBlock)) {
+    // TODO: NOTE: Added logic to make sure pressing enter is consistent between browsers
+    // As an example a fake caret is used before/after tables on FIrefox but not on Chrome so different behaviour could occur
+    // TODO: Create Jira to improve this
+    // const isBefore = parentBlock.getAttribute('data-mce-caret') === 'before';
+    // const node = isBefore ? parentBlock.nextElementSibling : parentBlock.previousElementSibling;
     newBlock = CaretContainer.showCaretContainerBlock(parentBlock) as Element;
     if (dom.isEmpty(parentBlock)) {
       NewLineUtils.emptyBlock(parentBlock);
     }
     NewLineUtils.setForcedBlockAttrs(editor, newBlock);
+    // if (isBefore && node) {
+    //   // Keep selection before the fake caret candidate
+    //   showCaret(-1, editor, node as HTMLElement, true, false).each((newRng) => {
+    //     moveToRange(editor, newRng);
+    //   });
+    // } else {
+    //   NewLineUtils.moveToCaretPosition(editor, newBlock);
+    // }
     NewLineUtils.moveToCaretPosition(editor, newBlock);
   } else if (isCaretAtStartOrEndOfBlock(false)) {
+    // Caret is moved to the new block in the insertNewBlockAfter fn
     newBlock = insertNewBlockAfter();
   } else if (isCaretAtStartOrEndOfBlock(true) && parentBlockParent) {
-    // Insert new block before
+    // Check where caret is positioned before it is potentially moved by 'insertBefore' fn
+    const caretPos = CaretPosition.fromRangeStart(rng);
+    const afterTable = isAfterTable(caretPos);
+    const parentBlockSugar = SugarElement.fromDom(parentBlock);
+    const afterBr = isAfterBr(parentBlockSugar, caretPos, editor.schema);
+    const prevBrOpt = afterBr
+      ? findPreviousBr(parentBlockSugar, caretPos, editor.schema).bind((pos) => Optional.from(pos.getNode()))
+      : Optional.none();
+
     newBlock = parentBlockParent.insertBefore(createNewBlock(), parentBlock);
-    const isNearChildren = Traverse.hasChildNodes(SugarElement.fromDom(rng.startContainer)) && rng.collapsed;
-    NewLineUtils.moveToCaretPosition(editor, (containerAndSiblingName(parentBlock, 'HR') || isNearChildren) ? newBlock : parentBlock);
+
+    const root = Optionals.someIf(containerAndSiblingName(parentBlock, 'HR') || afterTable, newBlock)
+      .or(prevBrOpt)
+      .getOr(parentBlock);
```

Comment:
The boolean could be calculated above the `insertBefore` fall, just to make absolutely clear that this doesn't depend on the result of `insertBefore` (although looking at the old code, perhaps it does?).

`someIf` is best used for standalone values. When you find you've used it to replace a conditional, just use a conditional :)
```suggestion
    const root = useNewBlock ? newBlock : preBrOpt.getOr(parentBlock);
```